### PR TITLE
Include SingleAtt in block flow aggregation, remove parallel flow

### PR DESCRIPTION
**Pull request by tbenr**
Due to not being a game-changer (and complex to implement) I decided to remove the parallel flow.

Added the single attestation only aggregation.
On healthy networks (mainnet), only considering aggregates we do not reach 8 attestations https://light-mainnet.beaconcha.in/slots/filtered?f&f.missing=1&f.orphaned=1&f.graffiti=TK
So I added a second pass of single attestation only aggregation, still limited by the aggregation time limit.

This also removes the `TimeLimitingIteretor` wrapper because introduces a weird edge case I fund during unit tests adjustments

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by mehdi-aouadi — APPROVED** [review]
LGTM

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/MatchingDataAttestationGroupV2.java [hunk 1]
```diff
@@ -414,6 +414,12 @@ private boolean noMatchingPreElectraAttestations(final Optional<UInt64> committe
   }
 
   private Supplier<Stream<PooledAttestation>> blockProductionCandidatesStreamSupplier() {
+    if (attestationsByValidatorCount.isEmpty()) {
+      // There are no aggregates left, which means they have all been included on-chain,
+      // so we can consider the long tail of single attestations that have not reached an aggregator
+      // in time
+      return () -> singleAttestationsByCommitteeIndex.values().stream().flatMap(Set::stream);
```
Comment (rolfyone) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/MatchingDataAttestationGroupV2.java:434]: these would be value based surely... do we sort them somewhere?

Comment (tbenr) [reply ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/MatchingDataAttestationGroupV2.java:434]: we do.
but i have another version locally, which is better than this simple one. I tested this and has the problem that we could start looking into SAs before completing the round on the aggregates.
